wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/OrganizationManager.java

```diff
@@ -0,0 +1,215 @@
+                               @Named("che.account.reserved_names") String[] reservedNames) {
+        this.organizationDao = organizationDao;
+        this.memberDao = memberDao;
+        this.eventService = eventService;
+        this.reservedNames = Sets.newHashSet(reservedNames);
+    }
+
+    /**
+     * Creates new organization
+     *
+     * @param newOrganization
+     *         organization to create
+     * @return created organization
+     * @throws NullPointerException
+     *         when {@code organization} is null
+     * @throws ConflictException
+     *         when organization with such id/name already exists
+     * @throws ConflictException
+     *         when specified organization name is reserved
+     * @throws ServerException
+     *         when any other error occurs during organization creation
+     */
+    public Organization create(Organization newOrganization) throws ConflictException, ServerException {
+        requireNonNull(newOrganization, "Required non-null organization");
+        if (reservedNames.contains(newOrganization.getName().toLowerCase())) {
+            throw new ConflictException(String.format("Organization name '%s' is reserved", newOrganization.getName()));
+        }
+        final OrganizationImpl organization = new OrganizationImpl(NameGenerator.generate("organization", 16),
+                                                                   newOrganization.getName(),
+                                                                   newOrganization.getParent());
+        organizationDao.create(organization);
+        eventService.publish(new OrganizationCreatedEvent(organization));
+        return organization;
+    }
+
+    /**
+     * Updates organization with new entity.
+     *
+     * @param organizationId
+     *         id of organization to update
+     * @param update
+     *         organization update
+     * @throws NullPointerException
+     *         when {@code organizationId} or {@code update} is null
+     * @throws NotFoundException
+     *         when organization with given id doesn't exist
+     * @throws ConflictException
+     *         when name updated with a value which is reserved or is not unique
+     * @throws ServerException
+     *         when any other error occurs organization updating
+     */
+    public Organization update(String organizationId, Organization update) throws NotFoundException,
+                                                                                      ConflictException,
+                                                                                      ServerException {
+        requireNonNull(organizationId, "Required non-null organization id");
+        requireNonNull(update, "Required non-null organization");
+        if (reservedNames.contains(update.getName().toLowerCase())) {
+            throw new ConflictException(String.format("Organization name '%s' is reserved", update.getName()));
```

Comment:
`create` and `update` methods share the same logic, it think this logic deserves a separate method, wdyt?